### DISCUSSION TITLE
Modular backup / Settings export & import

### DISCUSSION THREAD

**Discussion by serrq** [mullvad/mullvadvpn-app#9019] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

I want backup my servers list in order to easily restore it in a new device or send it to a friend.

### Alternative solutions

![1000118271](https://github.com/user-attachments/assets/6c816217-6bce-4e37-a704-e2fe0490d371)

### Type of feature

- [ ] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [X] iOS
- [X] Windows
- [X] macOS
- [X] Linux

**Comment by serrq** [comment]
Of course 100% offline. It should be a file like locations.zip.

**Comment by MrChocolatine** [comment]
It would nice to have this feature in the CLI:

```
mullvad custom-list export <LIST_NAME>
```

And when importing:

```
mullvad custom-list import <LIST_NAME> <PATH_FILE_TO_IMPORT>
```

**Comment by dlon** [comment]
Thanks for the suggestion. Our team has discussed adding a feature for exporting or importing custom lists before, though we have no plan to work on it yet.

On desktop, you can back up (and replace) your entire settings file, including custom lists, at `/etc/mullvad-vpn/settings.json`. This path varies between OSes (see https://github.com/mullvad/mullvadvpn-app?tab=readme-ov-file#file-paths-used-by-mullvad-vpn-app). Note that reloading your settings file requires restarting the Mullvad system service.

**Comment by serrq** [comment]
On Android?

I am not a developer.

**Comment by dlon** [comment]
@serrq It is not possible at the moment (without a lot of inconvenience).

**Comment by serrq** [comment]
Please, don't forget it 😭

**Comment by serrq** [comment]
Also settings preferences backup in modular manner like this app does (Libretube).

So a user can doing a tailored backup according with its needs.

![modular-backup](https://github.com/user-attachments/assets/549826a5-aa2d-4595-a33e-340cef2feec9)
